Google.Api.Generator.Rest/Models/MethodModel.cs

```diff
@@ -15,25 +15,84 @@
 using Google.Api.Generator.Utils;
 using Google.Api.Generator.Utils.Roslyn;
 using Google.Apis.Discovery.v1.Data;
+using Google.Apis.Services;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Generic;
+using System.Linq;
 using static Google.Api.Generator.Utils.Roslyn.RoslynBuilder;
 
 namespace Google.Api.Generator.Rest.Models
 {
     public class MethodModel
     {
+        private RestMethod _restMethod;
+
         public PackageModel Package { get; }
+        /// <summary>
+        /// Resource that owns this method, if any.
+        /// </summary>
+        public ResourceModel Resource { get; }
+        
+        public IReadOnlyList<ParameterModel> Parameters { get; }
+        public Typ ParentTyp { get; }
+        public Typ RequestTyp { get; }
         public string Name { get; }
 
-        public MethodModel(PackageModel package, string name, RestMethod restMethod)
+        public MethodModel(PackageModel package, ResourceModel resource, string name, RestMethod restMethod)
         {
             Package = package;
+            Resource = resource;
             Name = name.ToUpperCamelCase();
+            ParentTyp = resource?.Typ ?? package.ServiceTyp;
```

Comment:
Oh, I din't know that the methods could be defined outside of the resources, directly on the service, don't remember having seen that.

Reply:
I don't have any concrete examples of it at the moment... we might be able to remove it later.